src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -3023,14 +3023,24 @@ def check_nw_ip_flow(cmd, client, vm, watcher_rg, watcher_name, direction, proto
                      resource_group_name=None, nic=None, location=None):
     VerificationIPFlowParameters = cmd.get_models('VerificationIPFlowParameters')
 
-    local_ip_address, local_port = local.split(':')
-    remote_ip_address, remote_port = remote.split(':')
+    try:
+        local_ip_address, local_port = local.split(':')
+        remote_ip_address, remote_port = remote.split(':')
+    except:
+        raise CLIError("usage error: the format of the '--local' and '--remote' should be like x.x.x.x:port")
+
     if not is_valid_resource_id(vm):
+        if not resource_group_name:
+            raise CLIError("usage error: '--resource-group' required when use the name of the vm.")
+
         vm = resource_id(
             subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
             namespace='Microsoft.Compute', type='virtualMachines', name=vm)
 
     if nic and not is_valid_resource_id(nic):
+        if not resource_group_name:
+            raise CLIError("usage error: '--resource-group' required when use the name of the nic.")
```

Comment:
This should look like:
`usage error: --nic NAME --resource-group NAME | --nic ID`

Same with VM.